src/main.cpp

```diff
@@ -3661,13 +3661,34 @@ bool CheckColdStakeFreeOutput(const CTransaction& tx, const int nHeight)
         if (lastOut.nValue == 3 * COIN)
             return true;
 
+        // This could be a budget block.
+        if (Params().IsRegTestNet())
+            return false;
+
+        // if mnsync is incomplete, we cannot verify if this is a budget block.
+        // so we check that the staker is not transferring value to the free output
+        if (!masternodeSync.IsSynced()) {
+            // First try finding the previous transaction in database
+            CTransaction txPrev; uint256 hashBlock;
+            if (!GetTransaction(tx.vin[0].prevout.hash, txPrev, hashBlock, true))
+                return error("%s : read txPrev failed: %s",  __func__, tx.vin[0].prevout.hash.GetHex());
+            CAmount amtIn = txPrev.vout[tx.vin[0].prevout.n].nValue + GetBlockValue(nHeight - 1);
+            CAmount amtOut = 0;
+            for (unsigned int i = 1; i < outs-1; i++) amtOut += tx.vout[i].nValue;
```

Comment:
Have a point here. 

What could happen if MN payments spork is not active?. the `amtOut` will not be calculated properly and fail the check that is below (possibly stalling the chain).

Reply:
If `SPORK_8_MASTERNODE_PAYMENT_ENFORCEMENT` is disabled, any block payee is valid.
The block value is still the same (5 PIV), just the masternode part is "up for grabs"  by the staker.
Why not calculated properly?

Reply:
because the for loop is going up to `outs-1`, if there MN enforcement spork is disabled then it will not sum the last utxo.

Reply:
we are inside `if (outs >=3 && lastOut.scriptPubKey != tx.vout[outs-2].scriptPubKey) {}`